frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-utils.ts

```diff
@@ -47,36 +50,201 @@ export const isGCPWIFCluster = (
-  Proxy: InfraFeatures['proxy-aware'],
-  'proxy-aware': InfraFeatures.proxyAware,
-  FipsMode: InfraFeatures.fipsMode,
-  fips: InfraFeatures.fipsMode,
-  tlsProfiles: InfraFeatures.tlsProfiles,
-  cnf: InfraFeatures.cnf,
-  cni: InfraFeatures.cni,
-  csi: InfraFeatures.csi,
-  sno: InfraFeatures.sno,
-  TokenAuth: InfraFeatures.tokenAuth,
-  tokenAuthGCP: InfraFeatures.tokenAuthGCP,
-  [OperatorHubCSVAnnotationKey.disconnected]: InfraFeatures.disconnected,
-  [OperatorHubCSVAnnotationKey.fipsCompliant]: InfraFeatures.fipsMode,
-  [OperatorHubCSVAnnotationKey.proxyAware]: InfraFeatures.proxyAware,
-  [OperatorHubCSVAnnotationKey.cnf]: InfraFeatures.cnf,
-  [OperatorHubCSVAnnotationKey.cni]: InfraFeatures.cni,
-  [OperatorHubCSVAnnotationKey.csi]: InfraFeatures.csi,
-  [OperatorHubCSVAnnotationKey.tlsProfiles]: InfraFeatures.tlsProfiles,
-  [OperatorHubCSVAnnotationKey.tokenAuthAWS]: InfraFeatures.tokenAuth,
-  [OperatorHubCSVAnnotationKey.tokenAuthAzure]: InfraFeatures.tokenAuth,
-  [OperatorHubCSVAnnotationKey.tokenAuthGCP]: InfraFeatures.tokenAuthGCP,
+export const infrastructureFeatureMap = {
+  disconnected: InfrastructureFeature.Disconnected,
+  Disconnected: InfrastructureFeature.Disconnected,
+  Proxy: InfrastructureFeature.ProxyAware,
+  ProxyAware: InfrastructureFeature.ProxyAware,
+  'proxy-aware': InfrastructureFeature.ProxyAware,
+  FipsMode: InfrastructureFeature.FIPSMode,
+  fips: InfrastructureFeature.FIPSMode,
+  FIPS: InfrastructureFeature.FIPSMode,
+  tlsProfiles: InfrastructureFeature.TLSProfiles,
+  TLSProfiles: InfrastructureFeature.TLSProfiles,
+  tls: InfrastructureFeature.TLSProfiles,
+  TLS: InfrastructureFeature.TLSProfiles,
+  cnf: InfrastructureFeature.CNF,
+  CNF: InfrastructureFeature.CNF,
+  cni: InfrastructureFeature.CNI,
+  CNI: InfrastructureFeature.CNI,
+  csi: InfrastructureFeature.CSI,
+  CSI: InfrastructureFeature.CSI,
+  sno: InfrastructureFeature.SNO,
+  SNO: InfrastructureFeature.SNO,
+  tokenAuth: InfrastructureFeature.TokenAuth,
+  TokenAuth: InfrastructureFeature.TokenAuth,
+  tokenAuthGCP: InfrastructureFeature.TokenAuthGCP,
+  TokenAuthGCP: InfrastructureFeature.TokenAuthGCP,
+  [OLMAnnotation.Disconnected]: InfrastructureFeature.Disconnected,
+  [OLMAnnotation.FIPSCompliant]: InfrastructureFeature.FIPSMode,
+  [OLMAnnotation.ProxyAware]: InfrastructureFeature.ProxyAware,
+  [OLMAnnotation.CNF]: InfrastructureFeature.CNF,
+  [OLMAnnotation.CNI]: InfrastructureFeature.CNI,
+  [OLMAnnotation.CSI]: InfrastructureFeature.CSI,
+  [OLMAnnotation.TLSProfiles]: InfrastructureFeature.TLSProfiles,
+  [OLMAnnotation.TokenAuthAWS]: InfrastructureFeature.TokenAuth,
+  [OLMAnnotation.TokenAuthAzure]: InfrastructureFeature.TokenAuth,
+  [OLMAnnotation.TokenAuthGCP]: InfrastructureFeature.TokenAuthGCP,
+};
+export const normalizeInfrastructureFeature = (feature: string): InfrastructureFeature =>
+  infrastructureFeatureMap[feature] ?? feature;
+
+// TODO Replace with JSONSchema validation
```

Comment:
should there be tickets filed for these TODOs so they're remembered?

Reply:
https://issues.redhat.com/browse/CONSOLE-4281

Reply:
/lgtm